### DISCUSSION TITLE
Google Colab vs. R Studio

### DISCUSSION THREAD

**Discussion by ewhittemore987** [LiNk-NY/IntroR#48] [Polls]
@ralythompson24 : From a user standpoint, are there different scenarios where using Google Colab is more advantageous than using R Studio (or vice versa)? Do you have any personal preference with one over the other?

Any other students have real-world experience with one or both of these platforms in your job? Do you have a preference regarding which you use? And what is the reason for your preference?

Prior to this class, all of my R experience was in R Studio (all academic, no real-world experience...yet...). I really enjoyed learning the platform in a different aspect by using Google Colab. Even though the language was basically the same, seeing these activities in a different format made me think through the problems a bit more rather than rely on the previous experience I had in R Studio.

Just curious to hear about everyone else's opinion and/or experiences. I created a quick poll below and thought might be fun to quantify such a comparison :)

**Comment by ralythompson24** [comment]
I personally don't ever work in Google Colab for R. The main reason we use it in this course is because we are working from those UBC STAT 545 .ipynb worksheets, which can easily be opened in Google Colab. Google Colab is a popular way to run Python though. 

From my experience, the vast majority of people who code in R regularly use a local installation of RStudio on their desktop. Some people use the [Posit Cloud](https://posit.cloud/) version. 

The benefit of the cloud version of RStudio vs. the local version is that you are using a cloud-based computer processor. You can pay for more memory and more computing power if you use the cloud version, which is helpful for big data-based projects. 

For the standard non-big-data project the local installation of RStudio does just fine. I prefer to work with the version on my computer, it integrates nicely with my files and folders, including OneDrive and Dropbox.

**Reply by LiNk-NY** [reply]
@ewhittemore987 

Thanks for posing the question and thanks Rachel for your response. 

Prof. Levi Waldron and I started to use Google Colab because it has a much lower barrier to entry. Aside from its quirks, Google Colab can get students running pretty quickly without having to figure out how to install R and RStudio on their computers. As a teaching resource, it is very useful to remove the distractions that come with RStudio and to focus on the language itself. 

More recently, we can use the [Apporto](https://cuny.apporto.com/) service provided by CUNY to get a VM with RStudio installed. That VM has more resources than the Posit Cloud which is very limited on the free tier. These VMs are provided by the Amazon EC2 service and are pretty reliable.
